### ISSUE TITLE
Question: how to ignore healthcheck endpoint with otelmux

### ISSUE THREAD

**Issue by huydinhle** [open-telemetry/opentelemetry-go-contrib#896]
We are using otelmux to instrument our gorilla mux requests. Is there a way to ignore /health endpoint ?

**Comment by Aneurysm9** [comment]
There are a couple ways to handle this.  You can attach the middleware to a mux that doesn't route to the health check endpoint and then compose that with one that does the health check but doesn't have the middleware.  Alternately, you could implement a custom sampler that samples out health check requests.  Finally, the `otelhttp` instrumentation has a `Filter` type that can be used to filter requests that should not have spans created.  A PR moving that type (and its helper implementations) to a common location and integrating it with the `otelmux` instrumentation would probably be welcomed.

**Comment by jaredpiedt** [comment]
@Aneurysm9 I need similar functionality so I'm happy to work on a PR. Do you still prefer moving `otelhttp.Filter` to a different location or is it ok to use that type directly in the `otelmux` instrumentation?

**Comment by dmathieu** [comment]
We're starting to use the Filter (and other attributes) in several places (the grpc instrumentation is the most recent one).
So I think moving it to a general location would be welcomed.

**Comment by jaredpiedt** [comment]
@dmathieu thank you for the quick reply. I noticed that the [`otelgrpc.Filter`](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/df16f32df86b40077c9c90d06f33c4cdb6dd5afa/instrumentation/google.golang.org/grpc/otelgrpc/grpctrace.go#L39) and [`otelhttp.Filter`](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/df16f32df86b40077c9c90d06f33c4cdb6dd5afa/instrumentation/net/http/otelhttp/common.go#L42) types have different function signatures. Would you like to keep both types and move them to a central package or figure out a way to combine the types?

**Comment by dmathieu** [comment]
Now thinking about this, making them shared may not be so easy, especially due to subtle differences with method signatures as we have in grpc.
Happy to hear other opinions, but maybe we can start with a separate option for otelmux.

**Comment by jaredpiedt** [comment]
@dmathieu I agree with keeping the two separate. Are you opposed to just using `otelhttp.Filter` in the `otelmux` library? That would create the least amount of changes to the code base, but I wasn't sure if it was bad practice to share code between different exporter packages?

**Comment by dmathieu** [comment]
If that were to happen, I think it would be better to have a shared/base package.

**Comment by Rshep3087** [comment]
We we're looking at this also at my company. Has there been any movement on this issue? We can look at picking it up if not.

**Comment by dmathieu** [comment]
No, there hasn't been any movement yet.

**Comment by Rshep3087** [comment]
So looking to move the filter package out of otelhttp, internal directory seems to make the most sense. It only has a util package at the moment.

Once that is moved to a shared/base package, otelhttp should only need imports changed. otelmux will need the filters config similar to otelhttp and filter called on the ServeHTTP method.
